activerecord/test/cases/persistence_test.rb

```diff
@@ -1439,6 +1439,15 @@ def test_primary_key_stays_the_same
     assert_equal("id", ClothingItem.primary_key)
   end
 
+  def test_query_constraints_list_is_an_empty_array_if_primary_key_is_nil
+    klass = Class.new(ActiveRecord::Base) do
+      self.table_name = "developers_projects"
+    end
+
+    assert_nil klass.primary_key
+    assert_equal [], klass.query_constraints_list
```

Comment:
I think we should use `assert_empty klass.query_constraints_list` instead. What do you think?